src/fs/metadata_log_operations/write.hh

```diff
@@ -294,6 +320,7 @@ private:
 
             switch (append_result) {
             case metadata_log::append_result::TOO_BIG:
+                _metadata_log._cluster_allocator.free(cluster_id);
```

Comment:
Is this a bugfix or part of read-only mode, if the former it should not be in the same commit at least, maybe even a standalone PR or putting it into zpp_fs would be better.